src/main.rs

```diff
@@ -1,10 +1,21 @@
 use std::env;
-
+use std::io::{self, Write, BufWriter};
+use std::process;
 fn main() {
-    let arguments = env::args().collect::<Vec<_>>();
-    
-    match arguments.len() {
-        1 => loop { println!("y") },
-        _ => loop { println!("{}", arguments[1]) },
+    {
+        // Move all locals into a scope, so that `process::exit()` is called after all destructors.
+        let output = env::args()
+            .skip(1)
+            .next()
+            .map(|x| x + "\n")
+            .unwrap_or("y\n".to_owned());
```

Comment:
Allocation can be avoided here if not needed, with `.unwrap_or_else(|| "y\n".to_owned())`.
However, allocating is unnecessary in all cases, you could have one variable for a `String` and another one for `&str`, sometimes borrowing into the `String` variable.